java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightStream.java

```diff
@@ -207,6 +207,8 @@ public void close() throws Exception {
         } else {
           AutoCloseables.close(closeables);
         }
+        // Remove any metadata after closing to prevent negative refcnt
+        applicationMetadata = null;
```

Comment:
Should this go in the finally instead? I suppose if this throws the application is unlikely to close again.

Reply:
I put it here because if something throws and error before this, it's likely the buffer was not closed then. So at least this way, the buffer could be examined, but either way I don't think the application could recover.